core/vulkan/vk_api_timing_layer/cc/vk_api_emitter.inc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "core/vulkan/perfetto_producer/perfetto_data_source.h"
+#include "perfetto/tracing/core/data_source_config.h"
+#include "perfetto/base/time.h"
+
+namespace api_timing {
+
+template <typename T>
+VkApiEmitter<T>::VkApiEmitter() {
+  core::PerfettoProducer<T>::Get().RegisterEmitter(this);
+}
+
+template <typename T>
+VkApiEmitter<T>::~VkApiEmitter() {
+  core::PerfettoProducer<T>::Get().UnregisterEmitter(this);
+}
+
+template <typename T>
+void VkApiEmitter<T>::StartTracing() {
+  for (auto marker : debug_markers_) {
```

Comment:
Does this list expand forever?
What happens if the same handle gets more than one debug marker (with different names)
  - This can happen if you create an object, delete an object, create a NEW object that gets the same handle.

We will accumulate for ever, probably not what we want.

Reply:
done, PTAL